prereise/gather/griddata/hifld/data_access/load.py

```diff
@@ -43,19 +71,50 @@ def get_epa_ampd(path, year=2019):
     # Trim trailing slashes as necessary to ensure that join works
     path = path.rstrip("/\\")
 
-    data = pd.concat(
+    # Build cache path if necessary
+    if cache:
+        cache_dir = os.path.join(os.path.dirname(__file__), "cache")
+        os.makedirs(cache_dir, exist_ok=True)
+
+    data = {year: {state: {} for state in abv2state} for year in years}
```

Comment:
It might be clearer to use `dict()` instead of `{}` or we could do `dict.fromkeys(years, dict.fromkeys(abv2state, dict()))` as an alternative.

Reply:
I would prefer `{}` over `dict()`, but `dict.fromkeys` does seem nice when we have all values the same. How would you feel about `dict.fromkeys(years, dict.fromkeys(abv2state, {}))`?

Reply:
Wait, actually I think there's a problem with `dict.fromkeys` here. If the value to be passed is mutable, it's shared among all the keys!
```python
>>> seq = ["a", "b"]
>>> d = dict.fromkeys(seq, {})
>>> d
{'a': {}, 'b': {}}
>>> d["a"][1] = 2
>>> d
{'a': {1: 2}, 'b': {1: 2}}
```

Reply:
I guess we could do:
```python
from collections import defaultdict
data = defaultdict(lambda: defaultdict(dict))
```
That also lets us do assignment three-levels deep, like we're doing with `data[year][state][month_num] = df`:
```python
>>> from collections import defaultdict
>>> d = defaultdict(lambda: defaultdict(dict))
>>> d[1][2][3] = 4
>>> d
defaultdict(<function <lambda> at 0x0000025ECD3FCF70>, {1: defaultdict(<class 'dict'>, {2: {3: 4}})})
```

It's shorted, but I think less immediately clear what it's doing, compared to the dict comprehension.

Reply:
Yeah, it reminds me the trouble of having mutable defaults. I don't like the nested `defaultdict` either which can be filled with arbitrary values. Let's keep the comprehension then.